docs: add deploymentId config and clarify encryption key for self-hosting (#89795)
vercel/next.js: docs/01-app/02-guides/data-security.mdx

When **self-hosting** your Next.js application across multiple servers, each server instance may end up with a different encryption key, leading to potential inconsistencies.

To mitigate this, you can overwrite the encryption key using the `process.env.NEXT_SERVER_ACTIONS_ENCRYPTION_KEY` environment variable. Specifying this variable ensures that your encryption keys are persistent across builds, and all server instances use the same key.

> The key must be a base64-encoded value with a valid AES key length (16, 24, or 32 bytes). Next.js generates 32-byte keys by default. You can generate one using your platform's cryptographic tools, for example:

Reviewer: ```suggestion
The key must be a base64-encoded value whose decoded length matches a valid AES key size (16, 24, or 32 bytes). Next.js generates 32-byte keys by default. You can generate a compatible key using your platform’s cryptographic tools, for example:
```

```bash
openssl rand -base64 32
```

> This is an advanced use case where consistent encryption behavior across multiple deployments is critical for your application. You should consider standard security practices such as key rotation and signing. See the [Self-Hosting guide](/docs/app/guides/self-hosting#server-functions-encryption-key) for deployment-specific considerations.

Reviewer: ```suggestion
This is an advanced use case where consistent encryption behavior across multiple deployments is critical for your application. Follow standard security practices such as key rotation and signing. See the [Self-Hosting guide](/docs/app/guides/self-hosting#server-functions-encryption-key) for deployment-specific considerations.
```

When **self-hosting** your Next.js application across multiple servers, each server instance may end up with a different encryption key, leading to potential inconsistencies.

To mitigate this, you can overwrite the encryption key using the `process.env.NEXT_SERVER_ACTIONS_ENCRYPTION_KEY` environment variable. Specifying this variable ensures that your encryption keys are persistent across builds, and all server instances use the same key.

The key must be a base64-encoded value with a valid AES key length (16, 24, or 32 bytes). Next.js generates 32-byte keys by default. You can generate one using your platform's cryptographic tools, for example:

```bash
openssl rand -base64 32
```

> > **Good to know:** This environment variable must be available at both **build time** and **runtime** with the same value.

Reviewer 2: I don't think it needs to be made available at runtime. 
`next build` writes the keys to `.next/prerender-manifest.json` and loads them from there at runtime

Author: I think it goes to `.next/cache/.rscinfo` https://github.com/vercel/next.js/blob/c28948e0b3c199bff9123f91970906e80e4b1f08/packages/next/src/server/app-render/encryption-utils-server.ts#L11, which means that for `standalone` mode it wouldn't be available at runtime (from the cache folder)

I'll rephrase this though

Reviewer 2: That is incorrect.
It's always read from prerender-manifest.json in production
The file you linked is only called in
1. during `next build`
2. during dev

<img width="1584" height="330" alt="Bildschirmfoto 2026-02-11 um 11 33 14" src="https://github.com/user-attachments/assets/28a9bb05-4a73-427a-9365-4efcf3582d8d" />
